File: Chapter 04/JS/4.4 Anti-Patterns/Before-FatController_AnemicDomain/orderController.js
// Local files must start with './' to tell Node to look in 
// the current directory rather than node_modules.
const Order = require('./order');
const MyDbContext = require('./myDbContext');
const SmtpEmailService = require('./smtpEmailService');

/**
 * ANTI-PATTERN: THE FAT CONTROLLER.
 * Business logic is "leaking" into the transport layer.
 */
class OrderController {
    async createOrder(req, res) {
        const { items, customerType, customerEmail } = req.body;

        // 1. Validation Logic
        if (!items || items.length === 0) {
            return res.status(400).send("Order must have items.");
        }

        // 2 & 3. Business Logic (Calculation/Discount)
        let total = items.reduce((sum, i) => sum + i.price * i.qty, 0);
        if (customerType === "Gold") {
            total *= 0.9;
        }

        // 4. Data Access Logic
        const db = new MyDbContext();
        const order = new Order();
        order.total = total;
        await db.save(order);

        // 5. External Service Logic
        const mail = new SmtpEmailService();
        mail.send(customerEmail, "Order Confirmed!");

        res.status(200).json({ id: order.id });
    }
}

File: Chapter 04/Python/4.4_anti_patterns/before_fat_controller_anemic_domain/order_controller.py
# The filename (module) comes after 'from', 
# and the class name comes after 'import'.
from order import Order
from my_db_context import MyDbContext
from smtp_email_service import SmtpEmailService

class OrderController:
    """
    ANTI-PATTERN: THE FAT CONTROLLER.
    A ticking time bomb of mixed concerns.
    """
    def create_order(self, request):
        # 1. Validation Logic
        if not request.items:
            return "Error: Order must have items", 400

        # 2 & 3. Business Logic
        total = sum(i.price * i.qty for i in request.items)
        if request.customer_type == "Gold":
            total *= 0.9

        # 4. Data Access Logic
        db = MyDbContext()
        order = Order()
        order.total = total
        db.orders.add(order)
        db.commit()

        # 5. External Service Logic
        email_svc = SmtpEmailService()
        email_svc.send(request.customer_email, "Confirmed!")

        return {"id": order.id}, 200